_examples/thrift/main.go

```diff
@@ -127,6 +133,7 @@ type Const struct {
 // The grammar
 type Thrift struct {
 	Includes   []string     `{ "include" @String`
+	Comments   []*Comment   `  | @@`
```

Comment:
I was able to get comments inside of services only by using this grammar.  It is still a bit hacky since the comments will be stored separately from the functions, but they will still be in order for my purposes.

type Method struct {
	Function *Function `@@`
	Comment  *string   `| @Comment`
}

type Function struct {
	ReturnType  *Type         `@@`
	Name        string        `@Ident`
	Arguments   []*Argument   `"(" [ @@ { "," @@ } ] ")"`
	Throws      []*Throw      `[ "throws" "(" @@ { "," @@ } ")" ]`
	Annotations []*Annotation `[ "(" @@ { "," @@ } ")" ]`
}

Thanks for the Antlr post, I'll take a look and see how to improve.  Is there anything you want to see for this pull request now?